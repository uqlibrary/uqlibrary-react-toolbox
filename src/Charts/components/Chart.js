import React from 'react';
import {PropTypes} from 'prop-types';
import {findDOMNode} from 'react-dom';
import Highcharts from 'highcharts';
import 'highcharts-exporting';

class Chart extends React.Component {
    static propTypes = {
        chartOptions: PropTypes.object.isRequired,
        className: PropTypes.string
    };

    constructor(props) {
        super(props);
        this.chart = null;
    }

    componentDidMount() {
        if (this.refs && this.refs.chart) {
            this.chart = new Highcharts.Chart(
                findDOMNode(this.refs.chart),
                this.props.chartOptions
            );
        }
    }

    componentDidUpdate() {
        if (this.chart) {
            this.chart.update(this.props.chartOptions);
        }
    }

    componentWillUnmount() {
        if (this.chart) {
            this.chart.destroy();
        }
    }

    render() {
        if (window.matchMedia) {
            const mediaQueryList = window.matchMedia('print');
            mediaQueryList.addListener(() => {
                this.chart.reflow();
            });
        }
        return (
            <div className={this.props.className} ref="chart" />
        );
    }
}

export default Chart;


