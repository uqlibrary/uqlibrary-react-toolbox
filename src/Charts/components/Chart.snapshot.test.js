/* eslint-disable */

import React from 'react';
import {shallow, mount} from 'enzyme';
import toJson from 'enzyme-to-json';

import Chart from './Chart';

function setup(testProps, isShallow = false) {
    if (isShallow)
        return shallow(<Chart {...testProps} />);

    return mount(<Chart {...testProps} />);
}

describe('Chart snapshot tests', () => {
    it('it should mount empty chart component', () => {
        const app = setup({
            chartOptions: {
                title: {
                    text: null
                },
                chart: {
                    type: 'column'
                },
                xAxis: {
                    categories: [],
                    labels: {
                        rotation: -45,
                        y: 18
                    }
                },
                yAxis: {
                    min: 0,
                    title: {
                        text: 'Title'
                    },
                    stackLabels: {
                        enabled: true
                    }
                },
                legend: {
                    align: 'right',
                    verticalAlign: 'top',
                    x: -30,
                    y: -10,
                    floating: true,
                    shadow: false
                },
                tooltip: {

                },
                plotOptions: {
                    column: {
                        stacking: 'normal'
                    }
                },
                series: [{"name":"Journal Article","data":[]},{"name":"Conference Paper","data":[]},{"name":"Book Chapter","data":[]},{"name":"Book","data":[]},{"name":"Other","data":[]}]
            },
        });

        expect(toJson(app)).toMatchSnapshot();
    });
});

