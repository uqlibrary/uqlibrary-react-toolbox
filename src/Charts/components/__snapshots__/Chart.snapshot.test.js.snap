// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chart snapshot tests it should mount empty chart component 1`] = `
<Chart
  chartOptions={
    Object {
      "chart": Object {
        "type": "column",
      },
      "legend": Object {
        "align": "right",
        "floating": true,
        "shadow": false,
        "verticalAlign": "top",
        "x": -30,
        "y": -10,
      },
      "plotOptions": Object {
        "column": Object {
          "stacking": "normal",
        },
      },
      "series": Array [
        Object {
          "_colorIndex": 0,
          "data": Array [],
          "name": "Journal Article",
        },
        Object {
          "_colorIndex": 1,
          "data": Array [],
          "name": "Conference Paper",
        },
        Object {
          "_colorIndex": 2,
          "data": Array [],
          "name": "Book Chapter",
        },
        Object {
          "_colorIndex": 3,
          "data": Array [],
          "name": "Book",
        },
        Object {
          "_colorIndex": 4,
          "data": Array [],
          "name": "Other",
        },
      ],
      "title": Object {
        "text": null,
      },
      "tooltip": Object {},
      "xAxis": Object {
        "categories": Array [],
        "labels": Object {
          "rotation": -45,
          "y": 18,
        },
      },
      "yAxis": Object {
        "min": 0,
        "stackLabels": Object {
          "enabled": true,
        },
        "title": Object {
          "text": "Title",
        },
      },
    }
  }
>
  <div />
</Chart>
`;
